### PR TITLE
Kusto: Added more features such as: database operations, check cluste…

### PR DESCRIPTION
**Pull request by liatbezalel**
…r/database name availability, cluster - start/stop, event hub data connection, list skus, list resource skus.  Added new api version (features are identical to the older version).

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [ ] My spec meets the review criteria:
  - [ ] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [ ] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by annatisch** [comment]
This PR is a continuation of #3595 which has already received ARM sign-off.

**Comment by annatisch** [comment]
Also @liatbezalel - I see that there are now two new specs in this PR - preview and privatepreview.
What's the difference between them and does the 2nd spec also require ARM sign-off?

**Comment by liatbezalel** [comment]
@annatisch  they are both identical, so there is no need for ARM sign-off. (Also fixed the missing reference).

**Comment by annatisch** [comment]
Thanks @liatbezalel!
There are still some linter errors (ARM violations):
https://github.com/Azure/azure-rest-api-specs/pull/3895/checks?check_run_id=17718124#user-content-ARM-Errors

There's also a large number of model validation errors:
https://travis-ci.org/Azure/azure-rest-api-specs/jobs/429432043

**Comment by liatbezalel** [comment]
@annatisch everything is green now :)

**Comment by annatisch** [comment]
Thanks @liatbezalel - I will try to review this afternoon

**Comment by liatbezalel** [comment]
@annatisch ping

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json [hunk 1]
```diff
@@ -0,0 +1,2117 @@
...
+          "type": "string",
+          "description": "The event hub consumer group."
+        },
+        "tableName": {
+          "type": "string",
+          "description": "The table where the data should be ingested. Optionally the table information can be added to each message."
+        },
+        "mappingRuleName": {
+          "type": "string",
+          "description": "The mapping rule to be used to ingestion the data. Optionally the mapping information can be added to each message."
```
Comment (annatisch) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json]: "The mapping rule to be used to ingest the data"

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json [hunk 2]
```diff
@@ -0,0 +1,2117 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Kusto/clusters/{clusterName}/skus": {
```
Comment (annatisch) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json]: This operation should use the x-ms-pageable extension.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json [hunk 3]
```diff
@@ -0,0 +1,2117 @@
...
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Kusto/skus": {
```
Comment (annatisch) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json]: This operation should use the x-ms-pageable extension.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json [hunk 4]
```diff
@@ -0,0 +1,2117 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Kusto/clusters/{clusterName}/skus": {
+      "get": {
+        "tags": [
+          "Clusters"
+        ],
+        "operationId": "Clusters_Skus",
```
Comment (annatisch) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/2018-09-07-preview/kusto.json]: I think this needs a more descriptive operation ID - something like:
`Clusters_ListSkusByResourceGroup`

**Review by annatisch — CHANGES_REQUESTED** [review]
Looks pretty good - just a few minor things.
In addition to my review comments - a couple of general things:
- Please check the descriptions to capitalize abbreviations (e.g. URI, GUID, ID etc) and using capital letters at the start and period at the end.
- Please move the spec files into "stable" or "preview" subfolders. Please take a look at other service spec directory structure to see how this is done.

Thanks! :)

**Comment by liatbezalel** [comment]
thanks @annatisch, resolved all your comments.

**Comment by annatisch** [comment]
Thanks @liatbezalel!
It looks like you will need to update the relative references to the common types since the change in directory structure, e.g.:
https://github.com/Azure/azure-rest-api-specs/blob/4a44ff314057aaf471a4ce33b703fb130754eba5/specification/azure-kusto/resource-manager/Microsoft.Kusto/preview/2017-09-07-privatepreview/kusto.json#L1619

**Comment by liatbezalel** [comment]
@annatisch fixed :)

**Comment by liatbezalel** [comment]
@annatisch 
1. Fixed.
2. We are ok with that, it won't break anything.

**Review by annatisch — APPROVED** [review]

**Pull request merged**